### PR TITLE
[INLONG-3674][Manager] DataFlowInfo initialization support plugin

### LINKED ISSUE #3674: [Feature][Manager] Sort DataFlow initialization support plugin
### Description

_No response_

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
### Title Name: [INLONG-3674][Manager] DataFlowInfo initialization support plugin

Fixes #3674 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

Use CustomInfoFactory to create different component plugin
including class loading and type checking

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/thirdparty/sort/util/SinkInfoUtils.java [hunk 1]
```diff
@@ -56,12 +59,20 @@ public class SinkInfoUtils {
...
 
     /**
      * Create sink info for DataFlowInfo.
      */
     public static SinkInfo createSinkInfo(SourceResponse sourceResponse, SinkResponse sinkResponse,
-            List<FieldInfo> sinkFields) {
+            List<FieldInfo> sinkFields, Map<String, Object> properties) {
+
+        String loadType = String.valueOf(properties.get(KEY_TYPE_CUSTOM_SINK_INFO_GENERATOR));
+        if (!("null".equalsIgnoreCase(loadType))) {
```
Comment (luchunliang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/thirdparty/sort/util/SinkInfoUtils.java:71]: 为什么不直接判断null，要String.valueOf转成"null"

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/thirdparty/sort/util/SourceInfoUtils.java [hunk 2]
```diff
@@ -62,10 +69,19 @@ public static boolean isBinlogAllMigration(SourceResponse sourceResponse) {
      */
     public static SourceInfo createSourceInfo(PulsarClusterInfo pulsarCluster, String masterAddress,
             ClusterBean clusterBean, InlongGroupInfo groupInfo, InlongStreamInfo streamInfo,
-            SourceResponse sourceResponse, List<FieldInfo> sourceFields) {
+            SourceResponse sourceResponse, List<FieldInfo> sourceFields, Map<String, Object> properties) {
+
+        String loadType = String.valueOf(properties.get(KEY_TYPE_CUSTOM_SOURCE_INFO_GENERATOR));
+        if (!("null".equalsIgnoreCase(loadType))) {
```
Comment (luchunliang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/thirdparty/sort/util/SourceInfoUtils.java:75]: ditto

**Review by luchunliang — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/thirdparty/sort/util/SinkInfoUtils.java (hunk 1 above)
Comment (vernedeng) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/thirdparty/sort/util/SinkInfoUtils.java:71]: values in _properties_  are _Object_ format,  should convert it to String first.

**Review by wardlican — APPROVED** [review]

**Comment by healchow** [comment]
The plugin should move to `manager-plugins` module.

File: inlong-manager/manager-web/src/main/resources/application.properties [hunk 3]
```diff
@@ -55,8 +55,4 @@ common.http-client.connectionRequestTimeout=3000
 # Configure auth plugin
 inlong.auth.type=default
 # Configure dataflow generator plugin
-inlong.sort.serialinfo.type=default
-inlong.sort.transinfo.type=default
-inlong.sort.sinkinfo.type=default
-inlong.sort.fieldinfo.type=default
-inlong.sort.sourceinfo.type=default
\ No newline at end of file
+inlong.sort.type=default
```
Comment (dockerzhang) [inline inlong-manager/manager-web/src/main/resources/application.properties:58]: # Configure the plugin to generate data stream metadata
inlong.sort.metadata.type

**Comment by healchow** [comment]
The plugin function was already added in #4954.

**Pull request closed without merging**